python/lsst/ip/diffim/imageDecorrelation.py

```diff
@@ -223,21 +233,32 @@ def run(self, exposure, templateExposure, subtractedExposure, psfMatchingKernel,
         return pipeBase.Struct(correctedExposure=correctedExposure, correctionKernel=corrKern)
 
     @staticmethod
-    def _computeDecorrelationKernel(kappa, svar=0.04, tvar=0.04):
+    def _computeDecorrelationKernel(kappa, svar=0.04, tvar=0.04, preConvKernel=None):
         """! Compute the Lupton/ZOGY post-conv. kernel for decorrelating an
         image difference, based on the PSF-matching kernel.
         @param kappa  A matching kernel 2-d numpy.array derived from Alard & Lupton PSF matching
         @param svar   Average variance of science image used for PSF matching
         @param tvar   Average variance of template image used for PSF matching
+        @param preConvKernel If not None, then pre-filtering was applied
+            to science exposure, and this is the pre-convolution kernel.
         @return a 2-d numpy.array containing the correction kernel
 
         @note As currently implemented, kappa is a static (single, non-spatially-varying) kernel.
         """
         kappa = DecorrelateALKernelTask._fixOddKernel(kappa)
-        kft = scipy.fftpack.fft2(kappa)
-        kft = np.sqrt((svar + tvar) / (svar + tvar * kft**2))
-        pck = scipy.fftpack.ifft2(kft)
-        pck = scipy.fftpack.ifftshift(pck.real)
+        kft = np.fft.fft2(kappa)
+        kft2 = np.conj(kft) * kft
+        denom = svar + tvar * kft2
+        if preConvKernel is not None:
+            mk = DecorrelateALKernelTask._fixOddKernel(kft)
+            mk[mk < 1e-3] = 1e-3  # Psf should not be < 0; and messes up denominator
+            mk = np.fft.fft2(mk)
+            mk2 = np.conj(mk) * mk
+            denom = svar * mk2 + tvar * kft2
+            denom[np.abs(denom) < 1e-3] = 1e-3  # fixes some aliasing issues
```

Comment:
Is this `1e-3` intrinsically the same as the `1e-3` above for `mk` or an unrelated constant? If different, please define a new constant with documentation for why you chose the value.